rangeDS/fenwickTree.cpp

```diff
@@ -1,6 +1,16 @@
 struct fenwickTree {
 	vector<ll> bit;
 	fenwickTree(int n) : bit(n,0) {}
+	fenwickTree(const vector<int> & a) : bit(a.size()) {
+		bit[0] = a[0];
+		for (int i = 1; i < (int)a.size(); i++)
+			bit[i] = bit[i-1] + a[i];
+		for (int i = a.size() - 1; i > 0; i--) {
```

Comment:
Cast a.size to an int, otherwise this will underflow in the case where a is empty

Reply:
Then you'll get essentially a runtime error, it's a pretty nasty bug